api4/license.go

```diff
@@ -296,3 +299,94 @@ func getPrevTrialLicense(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	w.Write([]byte(model.MapToJSON(clientLicense)))
 }
+
+func requestTrueUpReview(c *Context, w http.ResponseWriter, r *http.Request) {
+	// Only admins can request a true up review.
+	if !c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PermissionManageSystem) {
+		c.SetPermissionError(model.PermissionManageLicenseInformation)
+		return
+	}
+
+	license := c.App.Channels().License()
+	if license == nil {
+		c.Err = model.NewAppError("requestTrueUpReview", "api.license.true_up_review.license_required", nil, "", http.StatusNotImplemented)
+		return
+	}
+
+	if license.IsCloud() {
+		c.Err = model.NewAppError("requestTrueUpReview", "api.license.true_up_review.not_allowed_for_cloud", nil, "", http.StatusNotImplemented)
+		return
+	}
+
+	profileMap, err := c.App.GetTrueUpProfile()
+	if err != nil {
+		c.Err = model.NewAppError("requestTrueUpReview", "api.license.true_up_review.get_status_error", nil, "", http.StatusInternalServerError)
+		return
+	}
+
+	profileMapJson, err := json.Marshal(profileMap)
+	if err != nil {
+		c.SetJSONEncodingError(err)
+		return
+	}
+
+	status, appErr := c.App.GetOrCreateTrueUpReviewStatus()
+	if err != nil {
+		c.Err = appErr
+		return
+	}
+
+	// Do not send true-up review data if the user has already requested one for the quarter.
+	// And only send a true-up review via as a one-time telemetry request if telemetry is disabled.
+	telemetryEnabled := c.App.Config().LogSettings.EnableDiagnostics
+	if !status.Completed && telemetryEnabled != nil && !*telemetryEnabled {
```

Comment:
Wondering if we should move the check of `status.Completed` earlier before getting into actually calculating all of the true up data and exist early if it has already completed? Technically it should never happen as I believe the front-end is not showing the panel if status is completed?

Reply:
Makes sense to do the completed check before most other conditions. I'll move it out of this one.